### PR TITLE
Updated SSD notebooks to collect data and use Azure ML

### PR DESCRIPTION
**Pull request by michhar**
Addition of 5 notebooks that take the user from collecting custom data directly from Percept DK, using Azure ML labeling projects, Azure ML to train, OpenVINO toolkit to convert and, finally, deploying through the portal (training and converting notebooks based on original SSD notebook).

New notebooks:

- 1.Collect_Image_Data_from_PerceptDK.ipynb
- 2.Upload_and_Label_with_AzureML.ipynb
- 3.Train_with_AzureML.ipynb
- 4.Convert_to_OpenVINO.ipynb
- 5.Deploy_to_PerceptDK.ipynb

@MaxStrange and team - feel free to provide comments, feedback, edits.  Thanks.

### CODE REVIEW

File: machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#############################################################################################
+# Copyright: Microsoft 2021
```
Comment (MaxStrange) [inline machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile:2]: Ugh. Technically we are supposed to put the following header at the top of every source code file in this repository:

```python
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT license.
```

But I'm not so sure that I have been making sure people have been doing that. Anyway, if you wouldn't mind, could you do that for your source files? Thaaaaaanks

File: machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+FROM nvidia/cuda:10.0-cudnn7-runtime
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update \
+      && apt-get install --no-install-recommends --no-install-suggests -y gnupg2 ca-certificates \
+            git build-essential libopencv-dev python3-opencv \
+      && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update \
```
Comment (MaxStrange) [inline machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile:20]: Is there are reason you split your RUN commands up here? I know it improves readability, but it also typically dramatically increases the size of Docker images.

Here's an interesting read about getting the size of Docker images down: https://learnk8s.io/blog/smaller-docker-images. I don't at all expect you to do more than just either use a multi-stage docker build or just combine some of these run statements. Or maybe you have a reason for having lots of run statements.

File: machine-learning-notebooks/transfer-learning-custom-azureml/1.Collect_Image_Data_from_PerceptDK.ipynb [hunk 3]
```diff
@@ -0,0 +1,354 @@
+{
```
Comment (MaxStrange) [inline machine-learning-notebooks/transfer-learning-custom-azureml/1.Collect_Image_Data_from_PerceptDK.ipynb:1]: I don't know how to leave a comment in a reasonable spot in a python notebook in GitHub reviews. But at some point in the notebook you refer to water bottles, but I think you are actually using office supplies as the example. I could be wrong.

File: machine-learning-notebooks/transfer-learning-custom-azureml/3.Train_with_AzureML.ipynb [hunk 4]
```diff
@@ -0,0 +1,1098 @@
...
+    "                                       'ssdlite_mobilenet_v2_coco_2018_05_09.tar.gz'))\n",
+    "move_config(config_fname)\n",
+    "# Train\n",
+    "train(config_fname, args.num_epochs)\n",
+    "# Convert TF checkpoint to a TF frozen graph for inferencing\n",
+    "export_frozen_graph(config_fname, model_path_prefix)\n",
+    "# Test on a single image from our data for sanity check\n",
+    "test_single_image(data_folder, num_classes=args.num_classes)\n",
+    "\n",
+    "# TODO\n",
```
Comment (MaxStrange) [inline machine-learning-notebooks/transfer-learning-custom-azureml/3.Train_with_AzureML.ipynb:927]: Do we want this TODO in here?

**Review by MaxStrange — CHANGES_REQUESTED** [review]
Awesome work, Micheleen! Thanks so much for putting this together; I know you are busy. Just a few minor changes and thoughts.

File: machine-learning-notebooks/transfer-learning-custom-azureml/1.Collect_Image_Data_from_PerceptDK.ipynb (hunk 3 above)
Comment (michhar) [reply machine-learning-notebooks/transfer-learning-custom-azureml/1.Collect_Image_Data_from_PerceptDK.ipynb:1]: Thanks for spotting.  Done.

File: machine-learning-notebooks/transfer-learning-custom-azureml/camera/capture_camera.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+from .camera import Camera
+import cv2
+import atexit
+import numpy as np
+import threading
+import traitlets
+
+
+class CaptureCamera(Camera):
```
[the remark this answers is not in the document]
Comment (michhar) [reply machine-learning-notebooks/transfer-learning-custom-azureml/camera/capture_camera.py]: Good point.  I've simplified this now with one class.

File: machine-learning-notebooks/transfer-learning-custom-azureml/datasets/dataset.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                self.annotations += [{
+                    'image_path': image_path,
+                    'category_index': category_index,
+                    'category': category
+                }]
+    
+    def save_entry(self, image, category):
+        """Saves an image in BGR8 format to dataset for category"""
+        if category not in self.categories:
+            raise KeyError('There is no category named %s in this dataset.' % category)
```
[the remark this answers is not in the document]
Comment (michhar) [reply machine-learning-notebooks/transfer-learning-custom-azureml/datasets/dataset.py]: Done.  Thanks.

File: machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile (hunk 1 above)
Comment (michhar) [reply machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile]: No problem!  Done.

File: machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile (hunk 2 above)
Comment (michhar) [reply machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile]: Brilliant, thanks!  My build was soooooooo much faster. :)

File: machine-learning-notebooks/transfer-learning-custom-azureml/3.Train_with_AzureML.ipynb (hunk 4 above)
Comment (michhar) [reply machine-learning-notebooks/transfer-learning-custom-azureml/3.Train_with_AzureML.ipynb]: Removed.  Good eye!

**Comment by michhar** [comment]
I've also updated the parent README.md to include a section on this notebook path.

**Review by MaxStrange — APPROVED** [review]

**Pull request merged**